code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1195,15 +1195,38 @@
 	if(SSpoints.xeno_points_by_hive[X.hivenumber] < psych_cost)
 		to_chat(owner, "<span class='xenowarning'>Someone used all the psych points while we were building!</span>")
 		return fail_activate()
+	building_turret(A, X)
+	SSpoints.xeno_points_by_hive[X.hivenumber] -= psych_cost
+	succeed_activate()
 
+/datum/action/xeno_action/activable/build_turret/proc/building_turret(atom/A, mob/living/carbon/xenomorph/X)
 	to_chat(owner, "<span class='xenowarning'>We build a new acid turret, spending 100 psychic points in the process</span>")
```

Comment:
This should probably be changed to reference `plasma_cost`, that way it's always consistent. (Same on line 1226 below)
```suggestion
	to_chat(owner, "<span class='xenowarning'>We build a new acid turret, spending [plasma_cost] psychic points in the process</span>")
```